lib/vmpooler/pool_manager.rb

```diff
@@ -555,6 +555,75 @@ def check_pool(pool,
       end
     end
 
+    def prepare_template(pool, provider)
+      if $config[:config]['create_template_delta_disks']
+        # Ensure templates are evaluated for delta disk creation on startup
+        return if $redis.hget('vmpooler__config__updating', pool['name'])
+        unless $redis.hget('vmpooler__template__prepared', pool['name'])
+          begin
+            $redis.hset('vmpooler__config__updating', pool['name'], 1)
+            provider.create_template_delta_disks(pool)
```

Comment:
this is open for multiple thread access (we have multiple threads per pool right?). My worry is that `vmpooler__config__updating` is not atomic and a race scenario is possible here. What's the worst case if a few threads run `create_template_delta_disks` in parallel?

If this is just to make sure we are not wasting resources it might be good enough to live with this exposure?

Reply:
In my testing I have seen that these sorts of safeguards successfully prevent duplicate instances of a task from running. In this context a delta disk is only being created for the template configured for the specific pool running the task. If two instances of this run I don't think there will be any issue as we can run create_delta_disks repeatedly without any discernible issue.